ios/Capacitor/Capacitor/CAPPluginMethod.swift

```diff
@@ -0,0 +1,17 @@
+//
+//  CAPPluginMethod.swift
+//  Capacitor
+//
+//  Created by Steven Sherry on 4/18/24.
+//  Copyright © 2024 Drifty Co. All rights reserved.
+//
+
+extension CAPPluginMethod {
+    public enum ReturnType: String {
+        case promise, callback
```

Comment:
should we add the `none` return type?

it's not used by any plugin other than console (as far as I know)

Reply:
We definitely could. none is weird in that the web code actually treats it as a callback when it is not a promise https://github.com/ionic-team/capacitor/blob/main/core/src/runtime.ts#L123-L133. The docs are actually misleading https://capacitorjs.com/docs/plugins/method-types#void-return in that is says "You can check the promise for an error but when it resolves the result is ignored" because you can't actually check for an error.

Reply:
let's add it, so if we want to do a refactor in the future for the core plugins, we can use this new initializer in all of them instead of having to use a different one for Console plugin